### PR TITLE
Close client twice (v3.0)

### LINKED ISSUE #55: Closing a client twice will block
@f2prateek 
@calvinfo 

Currently the `Close` method is not idempotent, and worst it blocks the program's execution if called more than once, see:
```go
// Close and flush metrics.
func (c *Client) Close() error {
    c.quit <- struct{}{}
    close(c.msgs)
    <-c.shutdown
    return nil
}
```
The issue is that nothing is reading from the `c.quit` channel anymore once the client has been closed.

One way of solving this is to close the `c.quit` channel from the client's goroutine before it terminates, but this would cause writing to the channel to panic on the second call of `Close`, so we have to protect this with a recover call, basically changing the method to something like this:
```go
// Close and flush metrics.
func (c *Client) Close() error {
    defer func() {
        // Always recover, a panic could be raised if c.quit was closed which means
        // the Close method was called more than once.
        recover()
    }()
    c.quit <- struct{}{}
    close(c.msgs)
    <-c.shutdown
    return nil
}
```
What do you guys think about it?

### PR DESCRIPTION
**Pull request by achille-roussel**
@f2prateek 
@calvinfo 

Fixes #55 

Follow up of #62 but will merge into `v3.0` instead of `master`.

Please let me know if anything needs to be changed.

### CODE REVIEW

File: analytics_test.go [hunk 1]
```diff
@@ -493,3 +493,27 @@ func TestTrackWithIntegrations(t *testing.T) {
...
+	if err := client.Close(); err != nil {
+		t.Error("closing a client should not a return an error")
+	}
+
+	if err := client.Close(); err != io.EOF {
+		t.Error("closing a client a second time should return io.EOF:", err)
+	}
+}
+
+func TestCloseTwice(t *testing.T) {
```
Comment (f2prateek) [inline analytics_test.go]: I think we can remove this one.

Comment (achille-roussel) [inline analytics_test.go]: Oops, copy/paste error... good catch.

**Comment by achille-roussel** [comment]
@f2prateek 

The tests are still failing with this trace:
```
godep: [WARNING]: godep should only be used inside a valid go package directory and
godep: [WARNING]: may not function correctly. You are probably outside of your $GOPATH.
godep: [WARNING]:	Current Directory: /home/ubuntu/analytics-go
godep: [WARNING]:	$GOPATH: /home/ubuntu/.go_workspace:/usr/local/go_workspace
# _/home/ubuntu/analytics-go/examples
examples/track.go:22: cannot use analytics.Track literal (type analytics.Track) as type *analytics.Track in argument to client.Track
ok  	_/home/ubuntu/analytics-go	1.145s	coverage: 62.2% of statements
godep: go exit status 2
make: *** [test] Error 1

make test returned exit code 2
```
Any idea what may be happening there? I do have the latest circle.yml file in this branch so I don't understand what's going on.

**Comment by achille-roussel** [comment]
Looks like it was a caching issue, Calvin helped me figuring it out.

**Comment by f2prateek** [comment]
cool LGTM!

**Pull request merged**